src/bridges/chainport.service.ts

```diff
@@ -0,0 +1,345 @@
+  is_lifi: boolean;
+};
+
+const chainportTokenSchema = Joi.object<ChainportToken>({
+  id: Joi.number().required(),
+  decimals: Joi.number().required(),
+  name: Joi.string().required(),
+  pinned: Joi.boolean().required(),
+  web3_address: Joi.string().required(),
+  symbol: Joi.string().required(),
+  token_image: Joi.string().required(),
+  chain_id: Joi.number().allow(null).required(),
+  network_name: Joi.string().required(),
+  network_id: Joi.number().required(),
+  blockchain_type: Joi.string().required(),
+  is_stable: Joi.boolean().required(),
+  is_lifi: Joi.boolean().required(),
+});
+
+const chainportTokenArraySchema =
+  Joi.array<ChainportToken[]>().items(chainportTokenSchema);
+
+type ChainportTokenListResponse = {
+  verified_tokens: ChainportToken[];
+};
+
+export type ChainportIronFishMetadata = {
+  bridge_output: {
+    publicAddress: string;
+    amount: string;
+    memoHex: string;
+    assetId: string;
+  };
+  gas_fee_output: {
+    publicAddress: string;
+    amount: string;
+    memo: string;
+  };
+  bridge_fee: {
+    source_token_fee_amount: string;
+    portx_fee_amount: string;
+    is_portx_fee_payment: boolean;
+  };
+};
+
+const chainportIronFishMetadataSchema = Joi.object<ChainportIronFishMetadata>({
+  bridge_output: Joi.object({
+    publicAddress: Joi.string().required(),
+    amount: Joi.string().required(),
+    memoHex: Joi.string().required(),
+    assetId: Joi.string().required(),
+  }).required(),
+  gas_fee_output: Joi.object({
+    publicAddress: Joi.string().required(),
+    amount: Joi.string().required(),
+    memo: Joi.string().required(),
+  }).required(),
+  bridge_fee: Joi.object({
+    source_token_fee_amount: Joi.string().required(),
+    portx_fee_amount: Joi.string().required(),
+    is_portx_fee_payment: Joi.boolean().required(),
+  }).required(),
+});
+
+export type ChainportPort =
+  | Record<string, never>
+  | {
+      base_network_id: number | null;
+      base_tx_hash: string | null;
+      base_tx_status: number | null;
+      base_token_address: string | null;
+      target_network_id: number | null;
+      target_tx_hash: string | null;
+      target_tx_status: number | null;
+      target_token_address: string | null;
+      created_at: string | null;
+      port_in_ack: boolean | null;
+    };
+
+@Injectable()
+export class ChainportService {
+  constructor(
+    private readonly config: ApiConfigService,
+    private readonly httpService: HttpService,
+    private readonly logger: LoggerService,
+  ) {}
+
+  private makeChainportRequest<T>(
+    url: string,
+  ): Promise<AxiosResponse<T, unknown>> {
+    return firstValueFrom(
+      this.httpService.get<T>(url).pipe(
+        catchError((e: AxiosError) => {
+          this.logger.error(
+            `Chainport Error: ${url} - Status Code ${
+              e.status ?? 'Unknown'
+            } - ${JSON.stringify(e.response?.data)}`,
+            new Error().stack ?? '',
+          );
+          throw e;
```

Comment:
NestJS has `BadGatewayException` if we want to throw a 502 here. I think if we see that in the responses then we can check the logs accordingly.